src/cli/load-compiler.ts

```diff
@@ -1,9 +1,9 @@
 import type { CompilerSystem } from '../declarations';
 
-export const loadCoreCompiler = async (sys: CompilerSystem): Promise<CoreCompiler> => {
-  await sys.dynamicImport(sys.getCompilerExecutingPath());
+export const loadCoreCompiler = async (_sys: CompilerSystem): Promise<CoreCompiler> => {
+  const coreCompiler = await import('@stencil/core/compiler');
 
-  return (globalThis as any).stencil;
+  return coreCompiler;
 };
```

Comment:
in an Esbuild world this actually doesn't need to be a dynamic import (`import` or `sys.dynamicImport`) at all - it can instead just be a normal `import` declaration, and as long as we have our `external` and `aliases` configurations set up correctly for Esbuild then it will compile down to either a `require` call or an `import` declaration, depending on the output format

I don't believe IE11 support (or in-browser support) is relevant here since we're in the CLI module, which I don't think we've ever supported using in a browser (unless I'm mistaken).

Reply:
Oh, right 🤦 NVM, carry on. Shoulda read the file name